### PR TITLE
added cached geocoder and test

### PR DESCRIPTION
**Pull request by jjangsangy**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,4 +11,4 @@ nltk
 numpy
 geopy
 reverse_geocoder
-scipy
\ No newline at end of file
+scipy
```
Comment (echu) [inline requirements.txt]: Why is scipy here?

File: src/geocoder/geocoder.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import functools
+
+from geopy.geocoders import GoogleV3
+
+def memoize(func):
```
Comment (echu) [inline src/geocoder/geocoder.py]: Maybe make this a utility function? Seems to be useful outside of this context.

File: src/geocoder/geocoder.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    return location.latitude, location.longitude
+
+
+google = GoogleV3()
+
+lat_long = functools.partial(get_lat_long, google)
+
+def run(node):
+    return {
```
Comment (echu) [inline src/geocoder/geocoder.py]: This is a neat one liner, but maybe make it cleaner / more readable? Specifically, we call `lat_long` twice.

File: src/geocoder/test_geocoder.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+from . geocoder import run
+
+@with_test(run, 'city')
+def data():
+    yield (
+        {'city': ('Portland, MA', 'Portland, OR')}, [
+        {'latitude': 42.077535,
+           'longitude': -72.654684,
+           'name': 'Portland, MA'},
+          {'latitude': 45.5230622, 'longitude': -122.6764816, 'name': 'Portland, OR'}
```
Comment (echu) [inline src/geocoder/test_geocoder.py]: Make the `{` consistently indented with the one above on L9.

File: src/geocoder/geocoder.py (hunk 2 above)
Comment (jjangsangy) [inline src/geocoder/geocoder.py]: Sure, should I add it to the `utility.py` module?

File: src/geocoder/geocoder.py (hunk 3 above)
Comment (jjangsangy) [inline src/geocoder/geocoder.py]: Okay we call lat_long only once now.

File: requirements.txt (hunk 1 above)
Comment (jjangsangy) [inline requirements.txt]: removed it

**Pull request merged**